pkg/service/active_directory_membership_service.go

```diff
@@ -0,0 +1,136 @@
+	return svc.objectSet.GetObjectListFromParams(params)
+}
+
+// GetActiveDirectoryMembershipsWithFields - method returns a array of pointers of type "ActiveDirectoryMembership" 
+func (svc *ActiveDirectoryMembershipService) GetActiveDirectoryMembershipsWithFields(fields []string) ([]*model.ActiveDirectoryMembership, error) {
+	params := &util.GetParams{}
+	params.WithFields(fields)
+	return svc.objectSet.GetObjectListFromParams(params)
+}
+
+// CreateActiveDirectoryMembership - method creates a "ActiveDirectoryMembership"
+func (svc *ActiveDirectoryMembershipService) CreateActiveDirectoryMembership(obj *model.ActiveDirectoryMembership) (*model.ActiveDirectoryMembership, error) {
+	// TODO: validate parameters
+	return svc.objectSet.CreateObject(obj)
+}
+
+// EditActiveDirectoryMembership - method modifies  the "ActiveDirectoryMembership" 
+func (svc *ActiveDirectoryMembershipService) EditActiveDirectoryMembership(id string, obj *model.ActiveDirectoryMembership) (*model.ActiveDirectoryMembership, error) {
+	return svc.objectSet.UpdateObject(id, obj)
+}
+
+// onlyActiveDirectoryMembership - private method for more than one element check. 
+func onlyActiveDirectoryMembership(objs []*model.ActiveDirectoryMembership) (*model.ActiveDirectoryMembership, error) {
+	if len(objs) == 0 {
+		return nil, nil
+	}
+
+	if len(objs) > 1 {
+		return nil, fmt.Errorf("More than one ActiveDirectoryMembership found with the given filter")
+	}
+
+	return objs[0], nil
+}
+
+ 
+// GetActiveDirectoryMembershipsByID - method returns associative a array of pointers of type "ActiveDirectoryMembership", filter by Id
+func (svc *ActiveDirectoryMembershipService) GetActiveDirectoryMembershipsByID(pool *model.Pool, fields []string) (map[string]*model.ActiveDirectoryMembership, error) {
+	params := &util.GetParams{}
+
+	// make sure ID field is selected
+	if _, found := params.FindField("id"); !found {
+		fields = append(fields, "id")
+	}
+	params.WithFields(fields)
+
+	// check if requested to filter by pool
+	if pool != nil {
+		filter := &util.SearchFilter{}
+		filter.Init("pool_id", util.EQUALS, *pool.ID, false)
+		params.WithSearchFilter(filter)
+	}
+	objs, err := svc.GetActiveDirectoryMemberships(params)
+	if err != nil {
+		return nil, err
+	}
+	objMap := make(map[string]*model.ActiveDirectoryMembership)
+	for _, obj := range objs {
+		objMap[*obj.ID] = obj
+	}
+	return objMap, nil
+}
+
+// GetActiveDirectoryMembershipById - method returns a pointer to "ActiveDirectoryMembership"
+func (svc *ActiveDirectoryMembershipService) GetActiveDirectoryMembershipById(id string) (*model.ActiveDirectoryMembership, error) {
+	return svc.objectSet.GetObject(id)
+}
+
+// GetActiveDirectoryMembershipsByName - method returns a associative array of pointers of type "ActiveDirectoryMembership", filter by name 
+func (svc *ActiveDirectoryMembershipService) GetActiveDirectoryMembershipsByName(pool *model.Pool, fields []string) (map[string]*model.ActiveDirectoryMembership, error) {
```

Comment:
I think we should drop these functions that return a map of id/name to the object.  It pollutes the SDK and is something we should allow the SDK users to implement rather than having these types of functions in all of the services.